### PR TITLE
run shellcheck

### LINKED ISSUE #64: Use shellcheck on all shell scripts
And also add a CI step that uses it!

There's been a couple of install script issues that could have been totally avoided. 

I'd also like to make sure the init scripts are correct

### PR DESCRIPTION
**Pull request by ismith**
Fixes #64

### CODE REVIEW

File: .github/workflows/shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Shellcheck
+
+on:
+  push:
+  pull_request:
```
Comment (conradludgate) [inline .github/workflows/shellcheck.yml:5]:
```suggestion
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```

**Comment by ismith** [comment]
I'm not sure why it's still not picking up the new action. Any ideas, @conradludgate ?

**Comment by conradludgate** [comment]
https://github.com/ellie/atuin/actions/runs/826177834/workflow

**Comment by ismith** [comment]
D'oh! That's it!

**Comment by ismith** [comment]
Ready for review. One thing to put eyes on: this https://github.com/ellie/atuin/pull/97/commits/92a74ea5550dcfb16078a2e94461c0b3d44fecf2 is fine, right? We're just checking for a status code there?

**Comment by bl-ue** [comment]
Shellcheck + zsh: https://github.com/koalaman/shellcheck/issues/809

**Comment by conradludgate** [comment]
Do you think you could add a comment to every ignore line?

File: src/shell/atuin.bash [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export ATUIN_SESSION=$(atuin uuid)
+ATUIN_SESSION=$(atuin uuid)
+export ATUIN_SESSION=
```
Comment (ellie) [inline src/shell/atuin.bash:2]:
```suggestion
export ATUIN_SESSION
```

**Review by ellie — CHANGES_REQUESTED** [review]
Thanks for this! I've just got one small suggestion (as it looks like the = [should not be there](https://github.com/koalaman/shellcheck/wiki/SC2155)?)

Happy to merge once this is done :blush:

**Comment by ismith** [comment]
Thanks for catching that! I also resolved the merge conflict in install.sh.

**Comment by ellie** [comment]
Thank you! :rocket:

**Review by ellie — APPROVED** [review]

File: install.sh [hunk 3]
```diff
@@ -138,16 +140,20 @@ __atuin_install_unsupported(){
...
 
-# TODO: Check which is in use
+# TODO: Check which shell is in use
+# Use of single quotes around $() is intentional here
+# shellcheck disable=SC2016
 printf '\neval "$(atuin init zsh)"' >> ~/.zshrc
 
 curl https://raw.githubusercontent.com/rcaloras/bash-preexec/master/bash-preexec.sh -o ~/.bash-preexec.sh
 printf '\n[[ -f ~/.bash-preexec.sh ]] && source ~/.bash-preexec.sh' >> ~/.bashrc
+# Use of single quotes around $() is intentional here
```
Comment (SuperSandro2000) [inline install.sh:157]: Yes, they are. We want the string as is into the rc file.

**Pull request merged**